src/main/java/org/terasology/las/yingyang/YingYangProvider.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.las.yingyang;
+
+import org.terasology.math.Region3i;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.world.generation.*;
+import org.terasology.world.generation.facets.SurfaceHeightFacet;
+import org.terasology.world.generator.plugin.RegisterPlugin;
+
+import java.util.Collection;
+import java.util.Collections;
+
+@RegisterPlugin
+@Produces(YingYangFacet.class)
+@Requires(@Facet(value = SurfaceHeightFacet.class, border = @FacetBorder(sides = 5)))
+public class YingYangProvider implements FacetProviderPlugin {
```

Comment:
This class is a provider for a _single YingYang pattern_ located at position (0,10,0), is that correct? Please note that in the JavaDoc for this class (and similarly for other "Singleton Facets" or whatever a good name would be for them...)